packages/subgraph/ponder.schema.ts

```diff
@@ -26,6 +42,24 @@ export const swap = onchainTable('swaps', (t) => ({
     createdAt: t.bigint(),
 }))
 
+// Denormalized events table for all analytics
+export const analyticsEvent = onchainTable('analytics_events', (t) => ({
+    id: t.text().primaryKey(), // ${txHash}-${logIndex}
+    spaceId: t.hex(),
+    eventType: t.text(), // 'swap', 'tip', etc.
+    blockTimestamp: t.bigint(),
+    txHash: t.hex(),
+
+    // ETH value for the event (calculated field for sorting/aggregation)
+    ethAmount: t.bigint().default(0n),
+
+    // Event-specific data stored as JSON
+    // For swap: { tokenIn, tokenOut, amountIn, amountOut, recipient, poster }
+    // For tip: { sender, receiver, currency, amount, tokenId, messageId, channelId }
+    // for join: { recipient, tokenId }
+    eventData: t.json(),
```

Comment:
Maybe we should type this a bit further that way we can easily add new events with some type safety in the future.

```
  type SwapEventData = {
      type: 'swap'
      tokenIn: string
      tokenOut: string
      amountIn: string
      amountOut: string
      recipient: string
      poster: string
  }

  type TipEventData = {
      type: 'tip'
      sender: string
      receiver: string
      currency: string
      amount: string
      tokenId: string
      messageId: string
      channelId: string
  }

  type JoinEventData = {
      type: 'join'
      recipient: string
      tokenId: string
  }

  type AnalyticsEventData = SwapEventData | TipEventData | JoinEventData

  // In the table:
  eventData: t.json<AnalyticsEventData>().notNull(),
```